run/jobs/main.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import random
+import sys
+import time
+
+# Retrieve Job-defined env vars
+TASK_NUM = os.getenv('TASK_NUM') or 0
+ATTEMPT_NUM = os.getenv('ATTEMPT_NUM') or 0
+# Retrieve User-defined env vars
+SLEEP_MS = os.getenv('SLEEP_MS')
+FAIL_RATE = os.getenv('FAIL_RATE')
+
+
+# Define main script
+def main(sleep_ms=None, fail_rate=None):
+    print(f"Starting Task #{TASK_NUM}, Attempt #{ATTEMPT_NUM}...")
+    # Simulate work
+    if (sleep_ms):
+        # Wait for a specific amount of time
+        time.sleep(float(sleep_ms)/1000)  # Convert to seconds
+
+    # Simulate errors
+    if (fail_rate):
+        try:
+            random_failure(float(fail_rate))
+        except Exception:
+            raise Exception(f"Task #{TASK_NUM}, Attempt #{ATTEMPT_NUM} failed.")
+
+    print(f"Completed Task #{TASK_NUM}.")
+
+
+# Throw an error based on fail rate
+def random_failure(rate):
+    if (rate is None or rate < 0 or rate > 1):
```

Comment:
I'm not sure you need to check for `None` here, even as written.  This function is only ever called after `if fail_rate`  -- so it should never be `None`.   

That being said, defaulting to 0 will better ensure that it is definitely never `None.`

Reply:
Also, please remove the parentheses!  https://google.github.io/styleguide/pyguide.html#33-parentheses